pkg/domains/netflow/netflow.go

```diff
@@ -0,0 +1,197 @@
+package netflow
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/korrel8r/korrel8r/internal/pkg/loki"
+	"github.com/korrel8r/korrel8r/pkg/domains/k8s"
+	"github.com/korrel8r/korrel8r/pkg/korrel8r"
+	"github.com/korrel8r/korrel8r/pkg/korrel8r/impl"
+	"github.com/korrel8r/korrel8r/pkg/openshift"
+	"golang.org/x/exp/maps"
+)
+
+var (
+	// Verify implementing interfaces.
+	_ korrel8r.Domain     = Domain
+	_ openshift.Converter = Domain
+	_ korrel8r.Store      = &store{}
+	_ korrel8r.Store      = &stackStore{}
+	_ korrel8r.Query      = Query{}
+	_ korrel8r.Class      = Class{}
+	_ korrel8r.Previewer  = Class{}
+)
+
+// Domain for log records produced by openshift-logging.
+//
+// There are several possible log store configurations:
+// - Default LokiStack store on current Openshift cluster: `{}`
+// - Remote LokiStack: `{ "lokiStack": "https://url-of-lokistack"}`
+// - Plain Loki store: `{ "loki": "https://url-of-loki"}`
+var Domain = domain{}
+
+type domain struct{}
+
+func (domain) Name() string                     { return "netflow" }
+func (d domain) String() string                 { return d.Name() }
+func (domain) Description() string              { return "Network flows from source nodes to destination nodes." }
+func (domain) Class(name string) korrel8r.Class { return Class{} }
+func (domain) Classes() []korrel8r.Class        { return []korrel8r.Class{Class{}} }
+func (d domain) Query(s string) (korrel8r.Query, error) {
+	_, s, err := impl.ParseQueryString(d, s)
```

Comment:
something I'm not sure to understand while reading the code is how LogQL is generated, I guess this is where it happens, but I don't know which value for `s` you typically get here.

In the end, for a given set of namespace+kind+pod, the LogQL that we want would be this: 

```
{SrcK8S_Namespace="my-namespace",SrcK8S_Type="Pod"} | json | SrcK8S_Name = "my-pod"
```

(or replace Pod with Service or Node when relevant, I guess this is the `class` returned by `ParseQueryString`)

Reply:
The LogQL is generated by rules - see korrel8r/etc/korrel8r/rules... which contain Go templates. The template takes data from some starting object, and constructs a query like you have written above.  Here's an example:
```
   - name: SelectorToLogs
     start:
       domain: k8s
       classes: [selectors]
     goal:
       domain: log
     result:
       query: |-
         log:{{k8sLogType .Namespace}}:{kubernetes_namespace_name="{{.Namespace}}"}|toJson
             {{- range $k,$v:=.Spec.Selector.MatchLabels}}|kubernetes_labels_{{lokiFixLabel $k}}="{{$v}}"{{end -}}
```
The `query` part is a template for a "korrel8r query" - which has the form `domain:class:somequerystring`, the format of the query string depends on the query domain - LogQL in this case. The example above iterates over the .Spec.Selector.MatchLabels of any k8s resource that has this field and constructs the LogQL equivalent. There are some helper functions provided by korrel8r (e.g. lokiFixLabels to sanitize selector keys) ,but most of it is just standard Go template on the Go struct that represents the resource.